Neos.Flow/Classes/Mvc/Controller/SpecialResponsesSupport.php

```diff
@@ -0,0 +1,93 @@
+<?php
+namespace Neos\Flow\Mvc\Controller;
+
+use Neos\Flow\Mvc\ActionRequest;
+use Neos\Flow\Mvc\ActionResponse;
+use Neos\Flow\Mvc\Exception\ForwardException;
+use Neos\Flow\Mvc\Exception\StopActionException;
+use Psr\Http\Message\UriInterface;
+
+/**
+ * provides helper methods to facilitate redirects, status throws, stop action and forwards
+ */
+trait SpecialResponsesSupport
+{
+    /**
+     * Sends the specified HTTP status immediately.
+     *
+     * @param integer $statusCode The HTTP status code
+     * @param string $content Body content which further explains the status the body of a given response will be overwritten if this is not empty
+     * @param ActionResponse|null $response The response to use or null for an empty response with the given status and message or content
+     * @return never
+     * @throws StopActionException
+     */
+    protected function responseThrowsStatus(int $statusCode, string $content = '', ?ActionResponse $response =  null): never
```

Comment:
do we want that people override those helpers? Otherwise they should be `final` wdym?

Reply:
right, no, no overwriting I think, you can just use your own code if you need some other behavior.